fs/zbd_zenfs.cc

```diff
@@ -86,14 +87,15 @@ uint64_t Zone::GetZoneNr() { return start_ / zbd_->GetZoneSize(); }
 void Zone::CloseWR() {
   assert(open_for_write_);
   Sync();
-
+  state_ = kReadOnly;
```

Comment:
I think the state changing should be placed inside the lock, just like how we change `open_for_write` flag.

Reply:
Good point. an atomic CAS will solve this problem.